rcl/test/rcl/test_client.cpp

```diff
@@ -129,6 +129,22 @@ TEST_F(TestClientFixture, test_client_init_fini) {
   EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string_safe();
   rcl_reset_error();
 
+  // Check if null publisher is valid
+  EXPECT_FALSE(rcl_client_is_valid(nullptr));
+  rcl_reset_error();
+
+  // Check if zero initialized client is valid
+  client = rcl_get_zero_initialized_client();
+  EXPECT_FALSE(rcl_client_is_valid(&client));
```

Comment:
Looks like the symbol `rcl_client_is_valid` is not being made visible on the dll on windows.
https://github.com/ros2/rcl/blob/38d3725262e25d5beb4c2b5c1155b74402d371d1/rcl/include/rcl/client.h#L391-L393

On windows symbols that need to be visible must have `__declspec(dllexport)` when the dll is being built, and `__declspec(dllimport)` when the dll is being used.

https://msdn.microsoft.com/en-us/library/a90k134d.aspx

`RCL_PUBLIC` macro takes care of this. Add it before the `rcl_*_is_valid()` function declarations and the linking error you're seen should go away.

Reply:
@sloretz thank you so much!